packages/eventual-send/src/E.js

```diff
@@ -53,11 +70,16 @@ function EProxyHandler(x, HandledPromise) {
 function EsendOnlyProxyHandler(x, HandledPromise) {
   return harden({
     ...baseFreezableProxyHandler,
-    get(_target, p, _receiver) {
-      return (...args) => {
-        HandledPromise.applyMethodSendOnly(x, p, args);
-        return undefined;
-      };
+    get(_target, p, receiver) {
+      return harden(
+        {
+          [p](...args) {
+            assert(this === receiver);
```

Comment:
This assertion appears to be the heart of this change.

Reply:
Correct, and the corresponding one above.